Lib/fontTools/ufoLib/glifLib.py

```diff
@@ -455,12 +457,10 @@ def writeGlyph(self, glyphName, glyphObject=None, drawPointsFunc=None, formatVer
 		fileName = self.contents.get(glyphName)
 		if fileName is None:
 			if self._existingFileNames is None:
-				self._existingFileNames = {}
-				for fileName in self.contents.values():
-					self._existingFileNames[fileName] = fileName.lower()
-			fileName = self.glyphNameToFileName(glyphName, self._existingFileNames.values())
+				self._existingFileNames = set(fileName.lower() for fileName in self.contents.values())
```

Comment:
maybe use a set comprehension instead? `{filename.lower() for ...}`

Reply:
granted, however making it an immutable set both in the default value (empty) and at the call site kills two or three birds: removes the need to use a mutable default or the check for None-ness

Reply:
ok, don't want to be too strict with immutability etc. Just treat (or annotate even) the `existing` parameter as a a `Container` (we just use its `__contains__` method) so it can accept any collections that is a container; then as default value use an immutable type so you don't have to check for none when called without explicit value; and at the call site, use a set comprehension.